tarantool/msgpack_ext/error.py

```diff
@@ -0,0 +1,52 @@
+"""
+Tarantool `error`_ extension type support module.
+
+Refer to :mod:`~tarantool.msgpack_ext.types.error`.
+
+.. _error: https://www.tarantool.io/en/doc/latest/dev_guide/internals/msgpack_extensions/#the-error-type
+"""
+
+from tarantool.types import (
+    encode_box_error,
+    decode_box_error,
+)
+
+EXT_ID = 3
+"""
+`error`_ type id.
+"""
+
+def encode(obj, packer):
+    """
+    Encode an error object.
+
+    :param obj: Error to encode.
+    :type obj: :class:`tarantool.BoxError`
+
+    :param packer: msgpack packer to encode error dictionary.
+    :type packer: :class:`msgpack.Packer`
+
+    :return: Encoded error.
+    :rtype: :obj:`bytes`
+    """
+
+    err_map = encode_box_error(obj)
+    return packer.pack(err_map)
```

Comment:
It feels inconsistency in `encode` implementations for extension types:

error - a separate function
datetime, interval - an unnecessary method
uuid - a field (implementation itself?)
decimal - implementation itself

You could unify it here before a release.

Reply:
> uuid - a field (implementation itself?)

It's "implementation itself": uuid just natively maps to byte sequence both with Tarantool and Python.

Reply:
> datetime, interval - an unnecessary method

Ok, let's make code more consistent. Since it would seem rather weird as a part of this PR, I have opened a separate one (which expected to be merged before this one): please, review #252

Reply:
Regarding
> error - a separate function
see https://github.com/tarantool/tarantool-python/pull/245#discussion_r1003072944 . This separate function does not create MP_EXT object. Moreover, there are cases when we need to decode only box.error dict without MP_EXT types (on the other way, there is no such cases for encode).

Reply:
#252 merged